mycroft/skills/skill_loader.py

```diff
@@ -150,7 +150,7 @@ def get_skill_class(skill_module):
     """
     for name, obj in skill_module.__dict__.items():
         if (isclass(obj) and issubclass(obj, MycroftSkill) and
-                obj is not MycroftSkill):
+                obj is not MycroftSkill and obj.is_compatible()):
```

Comment:
what if 2 compatible skills exist in same file? maybe one of them has less features but is still compatible

Reply:
Right, something like `Alternative Y is also compatible but will not be used`